python/grass/jupyter/interactivemap.py

```diff
@@ -347,13 +354,104 @@ def add_layer_control(self, **kwargs):
+            )
+            output_widget.layout.display = "block" if self.query_mode else "none"
+            # Change cursor style based on query mode
+            if self.query_mode:
+                self.map.default_style = {"cursor": "crosshair"}
+            else:
+                self.map.default_style = {"cursor": "default"}
+
+        query_button.on_click(toggle_query_mode)
+
+        # Add the button to the map using WidgetControl
+        query_control = self._ipyleaflet.WidgetControl(
+            widget=query_button, position="topright"
+        )
+        self.map.add_control(query_control)
+
+        # Create an output widget to display query results
+        output_control = self._ipyleaflet.WidgetControl(
+            widget=output_widget, position="bottomright"
+        )
+        self.map.add_control(output_control)
+
+        # Function to handle map click for querying
+        def handle_interaction(**kwargs):
+            if self.query_mode and kwargs.get("type") == "click":
+                lonlat = kwargs.get("coordinates")
+                reprojected_coordinates = reproject_latlon((lonlat[0], lonlat[1]))
+                raster_output = query_raster(
+                    (reprojected_coordinates[0], reprojected_coordinates[1]),
+                    self.raster_name,
+                )
+                vector_output = query_vector(
+                    (reprojected_coordinates[0], reprojected_coordinates[1]),
+                    self.vector_name,
+                )
+                with output_widget:
+                    output_widget.clear_output()
+                    output = (
+                        "<div style='font-family: Arial, sans-serif; font-size: 14px;'>"
+                    )
+                    if self.raster_name:
+                        output += "<b>Raster Info:</b><br>"
+                        for key, value in raster_output[0].items():
+                            output += f"&nbsp;&nbsp;Map: {key}<br>"
+                            for sub_key, sub_value in value.items():
+                                output += f"&nbsp;&nbsp;{sub_key}: {sub_value}<br>"
+                    if self.vector_name and len(vector_output[0]) > 2:
+                        output += "<b>Vector Info:</b><br>"
+                        for key, value in vector_output[0].items():
+                            if isinstance(value, dict):
+                                output += f"&nbsp;&nbsp;{key}:<br>"
+                                for sub_key, sub_value in value.items():
+                                    output += (
+                                        f"&nbsp;&nbsp;&nbsp;&nbsp;{sub_key}: "
+                                        f"{sub_value}<br>"
+                                    )
+                            else:
+                                output += f"&nbsp;&nbsp;{key}: {value}<br>"
+                    output += "</div>"
```

Comment:
The formatting should be a separate function, maybe part of the `query_raster` and `query_vector` functions.

Reply:
For raster, I think we are interested in the value only, not label or color.